woltka/workflow.py

```diff
@@ -0,0 +1,590 @@
+    using the latter.
+    """
+    if coords_fp:
+        click.echo(f'Reading gene coordinates...', nl=False)
+        with readzip(coords_fp) as fh:
+            coords = read_gene_coords(fh, sort=True)
+        click.echo(' Done.')
+        click.echo(f'Total host sequences: {len(coords)}.')
+        return Ordinal(coords, whether_prefix(coords),
+                       overlap and overlap / 100)
+    else:
+        return Plain()
+
+
+def build_hierarchy(names_fp:   str = None,
+                    nodes_fp:   str = None,
+                    newick_fp:  str = None,
+                    lineage_fp: str = None,
+                    ranktb_fp:  str = None,
+                    map_fps:   list = None,
+                    map_rank:  bool = False) -> (dict, dict, dict, str):
+    """Construct hierarchical classification system.
+
+    Parameters
+    ----------
+    names_fp : str, optional
+        Taxonomic names file.
+    nodes_fp : str, optional
+        Taxonomic nodes file.
+    newick_fp : str, optional
+        Newick tree file.
+    lineage_fp : str, optional
+        Lineage strings file.
+    ranktb_fp : str, optional
+        Rank table file.
+    map_fps : list of str, optional
+        Mapping files.
+    map_rank : bool, optional
+        Mapping filename is rank.
+
+    Returns
+    -------
+    dict, dict, dict, str
+        Taxonomic tree.
+        Rank dictionary.
+        Name dictionary.
+        Root identifier.
+    """
+    tree, rankd, named = {}, {}, {}
+    is_build = any([
+        names_fp, nodes_fp, newick_fp, lineage_fp, ranktb_fp, map_fps])
+    if is_build:
+        click.echo(f'Reading classification system...', nl=False)
+
+    # taxonomy names
+    if names_fp:
+        with readzip(names_fp) as f:
+            named = read_names(f)
+
+    # taxonomy nodes
+    if nodes_fp:
+        with readzip(nodes_fp) as f:
+            tree_, rankd_ = read_nodes(f)
+        update_dict(tree, tree_)
+        update_dict(rankd, rankd_)
+
+    # Newick-format tree
+    if newick_fp:
+        with readzip(newick_fp) as f:
+            update_dict(tree, read_newick(f))
+
+    # lineage strings file
+    if lineage_fp:
+        with readzip(lineage_fp) as f:
+            tree_, rankd_ = read_lineage(f)
+        update_dict(tree, tree_)
+        update_dict(rankd, rankd_)
+
+    # rank table file
+    if ranktb_fp:
+        with readzip(ranktb_fp) as f:
+            update_dict(tree, read_ranktb(f))
+
+    # plain mapping files
+    for fp in map_fps:
+        rank = path2stem(fp)  # filename stem as rank
+        with readzip(fp) as f:
+            map_ = read_map(f)
+        update_dict(tree, map_)
+        if map_rank:
+            update_dict(rankd, {k: rank for k in set(map_.values())})
+
+    # fill root
+    root = fill_root(tree)
+
+    if is_build:
+        click.echo(' Done.')
+        click.echo(f'Total classification units: {len(tree)}.')
+    return tree, rankd, named, root
+
+
+def reshape_rmap(rmap:    dict,
```

Comment:
```suggestion
def reshape_readmap(readmap:    dict,
```

Reply:
See below. Function name changed but variable name not.